core/server/data/migrations/utils.js

```diff
@@ -421,12 +421,63 @@ function createDropColumnMigration(table, column, columnDefinition) {
     );
 }
 
+/**
+ * Creates a migration which will insert a new setting in settings table
+ * @param {object} settingSpec - setting key, value, group and type
+ *
+ * @returns {Object} migration object returning config/up/down properties
+ */
+function addSetting({key, value, type, group}) {
+    return createNonTransactionalMigration(
```

Comment:
Why is this non transactional?

Reply:
@allouis Ah so this was picked up from the pattern we use for `createAddColumnMigration` which also used non transactional, but probably that's wrong here and we can/should use transactional instead ? 🤔

Reply:
Ah okay - so there's two types of migrations DML and DDL - essentially are we changing the data, or are we changing the tables. If we're changing the data then it should always be in a transaction (so that any errors are rolled back completely), if we're changing the tables (e.g. adding a column) then these cannot be done inside a transaction!

Hopefully that all makes sense

Reply:
This makes perfect sense! Thanks for the details 😄 I'll update this to wrap in a transaction

Reply:
Done ✅